### DISCUSSION TITLE
How can I get chat API endpoint call (to Update an existing Chat) to show up in the Chat UI?

### DISCUSSION THREAD

**Discussion by ksnyder2024** [open-webui/open-webui#3589] [Q&A]
Essentially I need to inject Messages from outside of the Chat UI into an existing Chat (to guide the User and provide text data). 
I've considered a few ideas but none of them have worked so far:
1. Submit a Broadcast message (something like role=broadcast), but there are no indications that is supported
2. If there's a browser refresh API call I can send (that would display the Message) that would work as well - but I've found no references to one
3. I've tried refreshing the browser, but that doesn't work
4. If there's a command I can submit to the LLM that would echo back to the User as a LLM message that would work as well (as long as I can trigger it from outside of the chat UI)

Since the POST chat update call is exposed at http://localhost:8080/api/v1/docs#/chats/update_chat_by_id_chats__id__post I'm hoping this is possible (otherwise why have it)?

I can Post a Request (the Body is below) that returns 200 OK, though the new message is not reflected in History (so I'm still not sure that the Request Body is correct). Unfortunately the FastAPI definition does not describe the messages element in the Chat Body - so I have no way to tell if my Request Body is correct except to look at the server output - and that appears to be correct except for the Request not being displayed in History.

My Request Body looks like this:
{
"chat": 
  {
    "messages": [
        {"parentId": "parent ID", "role": "user", "content": "Hello", "timestamp": timestamp from parent message, "models": ["llama3:latest"], "childrenIds: []}
    ]
  }
}

**Comment by TiFaBl** [comment]
I have a similar requirement where I want to inject additional data into the chat history without displaying it. Background is that we have developed a rather complex retrieval agent (using the Pipeline framework) and I want to make the full information (extensive SQL query returns) available to other models but not liter the chat with multiple pages of SQL and responses.

**Comment by m8dhouse** [comment]
to create a NEW chat I found out (trial and error) that this is the basics for the chats/new API call:
```

{
  "chat": 
{
    "models": [
      "gpt-4o-mini"
    ],
    "messages": [
      {
        "role": "user",
        "content": "\"wat is a podenco?\""
      }
    ]
}
}
```

You can add title etc - that also wiorks
So far it creates the chat, but I haven't managed that it also already executes it.

I started from the structure you get when you do a Chat Get API call and worked my way back:
```
{
  "id": "5edc69d9-0b47-4236-8529-bc8054640254",
  "user_id": "0f56e081-a505-4ada-8c1d-87f26e73e556",
  "title": "🐕 Podenco Dog Breed Overview",
  "chat": {
    "id": "",
    "title": "🐕 Podenco Dog Breed Overview",
    "models": [
      "gpt-4o-mini"
    ],
    "params": {},
    "messages": [
      {
        "id": "e0116b00-0b4a-4471-994f-921d63754c60",
        "parentId": null,
        "childrenIds": [
          "e8c63c32-2aaf-40ba-8a5d-a8dc940296b6"
        ],
        "role": "user",
        "content": "\"wat is a podenco\"",
        "timestamp": 1722436344,
        "models": [
          "gpt-4o-mini"
        ]
      },
      {
        "parentId": "e0116b00-0b4a-4471-994f-921d63754c60",
        "id": "e8c63c32-2aaf-40ba-8a5d-a8dc940296b6",
        "childrenIds": [],
        "role": "assistant",
        "content": "A Podenco is a breed of dog that originates from the Iberian Peninsula, particularly Spain and Portugal. These dogs are known for their hunting abilities, agility, and keen sense of smell. Podencos are often characterized by their slender bodies, long legs, large ears, and short or medium-length coats. They come in various sizes and can have different coat types, including smooth, rough, or long-haired. \n\nThere are several varieties of Podencos, including the Podenco Ibicenco, Podenco Canario, and Podenco Andaluz, each with unique traits and characteristics. Historically, they have been used for hunting rabbits and other small game, and they are known for their speed and endurance. Despite their hunting background, Podencos can make loving and loyal companions when properly socialized and trained.",
        "model": "gpt-4o-mini",
        "modelName": "gpt-4o-mini",
        "userContext": null,
        "timestamp": 1722436344,
        "lastSentence": "Historically, they have been used for hunting rabbits and other small game, and they are known for their speed and endurance.",
        "done": true
      }
    ],
    "history": {
      "messages": {
        "e0116b00-0b4a-4471-994f-921d63754c60": {
          "id": "e0116b00-0b4a-4471-994f-921d63754c60",
          "parentId": null,
          "childrenIds": [
            "e8c63c32-2aaf-40ba-8a5d-a8dc940296b6"
          ],
          "role": "user",
          "content": "\"wat is a podenco\"",
          "timestamp": 1722436344,
          "models": [
            "gpt-4o-mini"
          ]
        },
        "e8c63c32-2aaf-40ba-8a5d-a8dc940296b6": {
          "parentId": "e0116b00-0b4a-4471-994f-921d63754c60",
          "id": "e8c63c32-2aaf-40ba-8a5d-a8dc940296b6",
          "childrenIds": [],
          "role": "assistant",
          "content": "A Podenco is a breed of dog that originates from the Iberian Peninsula, particularly Spain and Portugal. These dogs are known for their hunting abilities, agility, and keen sense of smell. Podencos are often characterized by their slender bodies, long legs, large ears, and short or medium-length coats. They come in various sizes and can have different coat types, including smooth, rough, or long-haired. \n\nThere are several varieties of Podencos, including the Podenco Ibicenco, Podenco Canario, and Podenco Andaluz, each with unique traits and characteristics. Historically, they have been used for hunting rabbits and other small game, and they are known for their speed and endurance. Despite their hunting background, Podencos can make loving and loyal companions when properly socialized and trained.",
          "model": "gpt-4o-mini",
          "modelName": "gpt-4o-mini",
          "userContext": null,
          "timestamp": 1722436344,
          "lastSentence": "Historically, they have been used for hunting rabbits and other small game, and they are known for their speed and endurance.",
          "done": true
        }
      },
      "currentId": "e8c63c32-2aaf-40ba-8a5d-a8dc940296b6"
    },
    "tags": [],
    "timestamp": 1722436344865,
    "files": []
  },
  "updated_at": 1722436348,
  "created_at": 1722436344,
  "share_id": null,
  "archived": false
}
```



**You can also use this url with parameters to create a new chat with content: (but not an API call)**
 <yoururl>?/?q=Query here&models=gpt-4o-mini

**Reply by MetaLab-UAutonoma** [reply]
When I call it I get  "405 Method Not Allowed". Did it happen to you? If so, How do you solve it?
I do have my api key and I have my sk- bearer token configured.
I'm using postman.

**Reply by MetaLab-UAutonoma** [reply]
Nevermind. I was missing "/api/v1"
